src/announcement/AnnouncementController.ts

```diff
@@ -0,0 +1,176 @@
+    public router: Router = Router();
+
+    constructor() {
+        debugger;
+        this.initializeRoutes();
+    }
+
+    private initializeRoutes() {
+        this.router.post(`${this.path}`, validationMiddleware(AddAnnouncementRequestDto), this.addAnnouncement);
+        this.router.get(`${this.path}/:id`, this.getAnnouncementById);
+        this.router.get(`${this.path}s/`, this.getAllAnnouncements);
+        this.router.put(`${this.path}/:id`, validationMiddleware(UpdateAnnouncementRequestDto), this.updateAnnouncement);
+        this.router.put(`${this.path}/:id/status`, validationMiddleware(ChangeAnnouncementStatusRequestDto), this.updateAnnouncementStatus);
+    }
+
+    private async getAllAnnouncements(request: Request, response: Response, next: NextFunction) {
+        const anncouncementId : string = request.params.id;
+        const tags = request.query.tags + "";
+        const parsed = tags.split(",").map(elem => elem);
+
+        try {
+            const tags: Tag[] | null = await tagRepository.createQueryBuilder("tag")
+            .where("tag.name IN (:...names)", { names: parsed})
+            .getMany(); 
+            
+            const announcements: Announcement[] | null = await announcementRepository.find({
+                where: {
+                    tags: tags
+                },
+                relations: {
+                    messages: true,
+                    tags: true,
+                }
+            })
+            if (announcements === null || announcements.length === 0) {
+                throw new AnnouncementNotFoundException(anncouncementId);
+            }
+            response.status(200).json(announcements);
+        } catch(error) {
+            next(error);
+        }    
+    }
+
+    private async getAnnouncementById(request: Request, response: Response, next: NextFunction) {
+        const anncouncementId : string = request.params.id;
+        try {
+            const announcement: Announcement | null = await announcementRepository.findOneBy({id: anncouncementId});
+            if (announcement === null) {
+                throw new AnnouncementNotFoundException(anncouncementId);
+            }
+            response.status(200).json(announcement);
+        } catch(error) {
+            next(error);
+        }    
+    }
+
+    private async updateAnnouncementStatus(request: Request, response: Response, next: NextFunction) {
+        const announcementData: ChangeAnnouncementStatusRequestDto = request.body as ChangeAnnouncementStatusRequestDto;
+        const anncouncementId : string = request.params.id;
+        try {
+
+            const announcement: Announcement | null = await announcementRepository.findOneBy({id: anncouncementId});
+            if (announcement === null) {
+                throw new AnnouncementNotFoundException(anncouncementId);
+            }
+            
+            announcement.isActive = announcementData.isActive || announcement.isActive;
```

Comment:
Try to split long lines into multiple rows 🙏

Reply:
can you use ?Announcement for nullable variables